virttest/utils_net.py

```diff
@@ -3196,45 +3197,47 @@ def get_linux_ipaddr(session, nic):
     """
     Get IP addresses by nic name
     """
+    rex = r"inet6?\s+(\S+)\s+"
     cmd = "ifconfig %s || ip address show %s" % (nic, nic)
-    out = session.cmd_output(cmd)
-    inet_regex = "inet (addr:\s?)?(\d+.\d+.\d+.\d+)"
-    try:
-        inet = re.search(inet_regex, out, re.M).groups()[1]
-    except Exception:
-        inet = None
-
-    inet6_regex = "inet6 (addr:\s?)?(\S+)"
-    try:
-        inet6 = re.search(inet6_regex, out, re.M).groups()[1].split('/')[0]
-    except Exception:
-        inet6 = None
-    return (inet, inet6)
+    out = session.cmd_output_safe(cmd)
+    addrs = re.findall(rex, out, re.M)
+    addrs = map(lambda x: x.split('/')[0], addrs)
+    addrs = map(lambda x: netaddr.IPAddress(x), addrs)
+    ipv4_addr = filter(lambda x: x.version == 4, addrs)
+    ipv6_addr = filter(lambda x: x.version == 6, addrs)
+    ipv4_addr = str(ipv4_addr[0]) if ipv4_addr else None
+    ipv6_addr = str(ipv6_addr[0]) if ipv6_addr else None
+    return (ipv4addr, ipv6addr)
 
 
 def windows_mac_ip_maps(session):
     """
     Windows get MAC IP addresses maps
     """
+    def str2ipaddr(str_ip):
+        try:
+            return netaddr.IPAddress(str_ip)
+        except Exception:
+            pass
+        return None
+
     maps = {}
     cmd = "wmic nicconfig where IPEnabled=True get ipaddress, macaddress"
     out = session.cmd_output(cmd)
+    regex = r".*\w{2}[:-]\w{2}[:-]\w{2}[:-]\w{2}[:-]\w{2}[:-]\w{2}\s*"
     for line in out.splitlines():
-        try:
-            mac = re.search("\w{2}:\w{2}:\w{2}:\w{2}:\w{2}:\w{2}", line).group(0)
-        except Exception:
+        if not re.match(regex, line):
             continue
-        inets = re.findall("\"([\w.:]*)\"", line)
-        try:
-            inet = inets[0]
-        except IndexError:
-            inet = None
-        try:
-            inet6 = inets[1]
-        except IndexError:
-            inet6 = None
-        maps[mac] = (inet, inet6)
-
+        line = re.sub(r"[\"\'\{\}]", "", line)
+        addr_info = map(lambda x: x.strip(), line.split(","))
+        mac = addr_info.pop().lower().replace("-", ":")
```

Comment:
I'm afraid the above codes can not correctly get the mac address, at least for win7.
```powershell
C:\> wmic nicconfig where IPEnabled=True get ipaddress, macaddress
IPAddress                                         MACAddress         
{"192.168.122.172", "fe80::6070:31b0:523e:cf5f"}  52:54:00:24:52:A1
```